src/main/java/seedu/address/model/Statistics.java

```diff
@@ -23,33 +28,120 @@ public Statistics(ReadOnlyStatistics toBeCopied) {
     public void resetData(ReadOnlyStatistics newData) {
         requireNonNull(newData);
         setDayDatas(newData.getDayDataList());
     }
 
-    //// person-level operations
+    //// dayData-level operations
+
+    // FUNCTIONS FOR
+    // HARDOHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH SIR
+
+    /** reinitialises dayDataList to current day while retaining stored data. */
+    public void fetch() {
+        HashMap<Date, PomDurationData> pomTempStorage = new HashMap<>();
+        HashMap<Date, TasksDoneData> tasksTempStorage = new HashMap<>();
+
+        for (int i = 0; i < dayDataList.size(); i++) {
+            pomTempStorage.put(
+                    dayDataList.get(i).getDate(), dayDataList.get(i).getPomDurationData());
+            tasksTempStorage.put(
+                    dayDataList.get(i).getDate(), dayDataList.get(i).getTasksDoneData());
+        }
+
+        this.init();
+
+        for (int i = 0; i < dayDataList.size(); i++) { // updated dayDatalist
+            Date currTempDate = dayDataList.get(i).getDate();
+            if (pomTempStorage.containsKey(currTempDate)) {
+                PomDurationData tempPomDurationDate = pomTempStorage.get(currTempDate);
+                TasksDoneData tempTasksDoneData = tasksTempStorage.get(currTempDate);
+                DayData tempDayData =
+                        new DayData(currTempDate, tempPomDurationDate, tempTasksDoneData);
+                dayDataList.remove(i);
+                dayDataList.add(i, tempDayData);
+            }
+        }
+
+        assert (dayDataList.size() <= MAX_SIZE);
+    }
+
+    /**
+     * Auto-checks and replaces a new DayData object at the same Date
+     *
+     * @param dayData
+     */
+    public void updatesDayData(DayData dayData) {
+        requireNonNull(dayData);
+
+        Date currDate = dayData.getDate();
+        for (int i = 0; i < dayDataList.size(); i++) {
+            DayData currDayData = dayDataList.get(i);
+            Date currDayDataDate = currDayData.getDate();
+            if (currDayDataDate.equals(currDate)) { // correct date
+                dayDataList.remove(i);
+                dayDataList.add(i, dayData);
+                return;
+            }
+        }
+        assert (false); // dayData not found
+    }
 
     /**
-     * Returns true if a person with the same identity as {@code person} exists in the address book.
+     * Gets DayData object at current date.
+     *
+     * @param date
      */
-    public boolean hasTask(DayData dayData) {
+    public DayData getDayDataFromDate(Date date) {
+        requireNonNull(date);
+
+        for (int i = 0; i < dayDataList.size(); i++) {
+            DayData currDayData = dayDataList.get(i);
+            Date currDayDataDate = currDayData.getDate();
+            if (currDayDataDate.equals(date)) {
+                return currDayData;
+            }
+        }
+
+        return null;
```

Comment:
would throwing an exception be better? hahah i have some parts where i returned null, but at least with the exceptions we can be more explicit about the error

Reply:
yup will update this!